File: CS/InnerClasses.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;

namespace DataContextWpfSample {

    public class DataStore : FrameworkElement {
        public static readonly DependencyProperty ConnectionStringProperty =
            DependencyProperty.Register("ConnectionString", typeof(string), typeof(DataStore), new PropertyMetadata(string.Empty));

        public string ConnectionString {
            get { return (string)GetValue(ConnectionStringProperty); }
            set { SetValue(ConnectionStringProperty, value); }
        }
    }
}

File: VB/InnerClasses.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows

Namespace DataContextWpfSample

	Public Class DataStore
		Inherits FrameworkElement
		Public Shared ReadOnly ConnectionStringProperty As DependencyProperty = DependencyProperty.Register("ConnectionString", GetType(String), GetType(DataStore), New PropertyMetadata(String.Empty))

		Public Property ConnectionString() As String
			Get
				Return CStr(GetValue(ConnectionStringProperty))
			End Get
			Set(ByVal value As String)
				SetValue(ConnectionStringProperty, value)
			End Set
		End Property
	End Class
End Namespace
